File: backend/src/models/feedback.py
from pydantic import BaseModel
from src.db.mongodb import get_collection
from typing import Optional

Feedbacks = get_collection("feedback")


class FeedbackPayload(BaseModel):
    rating: int
    feedbackType: str
    comment: str
    reccomendation: str
    webId: Optional[str] = None
    userId: Optional[str] = None

File: frontend/src/hooks/feedback.ts
import { useMutation } from "@tanstack/react-query";
import { api } from "@/lib/api";

type FeedbackPayload = {
    rating: number;
    feedbackType: string;
    comment: string;
    reccomendation: string;
    webId?: string | null;
    userId?: string | null; 
};

export const useSendFeedback = () => {
  return useMutation({
    mutationFn: async (config: FeedbackPayload) => {
      const response = await api.post("/feedback", config);
      return response.data.result;
    },
    onSuccess: () => {},
    onError: () => {},
  });
};
